### PR TITLE
[Woo POS] Fix stores without the supported country/currency can still access POS

### LINKED ISSUE #14367: [Woo POS] Stores not in the supported country (the US) can still access POS
**Describe the bug**

Stores not set in the US can still see the POS entry point in the Menu tab.

**To Reproduce**
- In wp-admin WC settings, set the country to be not in the US while the currency is USD
- In the app, go the Menu --> notice that shortly, the POS row still appears
- Tap on `Connect your reader` --> an error about the country not supported is shown

**Screenshots**

https://github.com/user-attachments/assets/f55bd9fc-e200-41d0-9e11-2c1056093d62

**Expected behavior**

**Isolating the problem (mark completed items with an [x]):**
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Mobile Environment**
Please include:
- Device: any iPad
- iOS version: any 
- WooCommerce iOS version: 21.1

### PR DESCRIPTION
**Pull request by jaclync**
Closes: #14367 

## Why

When the POS payments onboarding feature flag is enabled, the site checks (country and currency) function isn't invoked because it's currently under the onboarding state publisher. Therefore, stores with unsupported country/currency can still access POS when the onboarding feature flag is enabled (false in production, so the issue isn't expected to affect production users).

## How

First, test cases were added/updated so that the new cases with the payments onboarding feature flag fail without any changes. Then, the site checks are included as a filter on the returned published whenever the WC version or the remote feature flag changes.

## Steps to reproduce

- In wp-admin WC settings, set the country to be not in the US while the currency is USD
- In the app, go to Menu --> the POS row should **not** appear. before this PR, the POS row appears shortly
- In wp-admin WC settings, set the country back to the US
- Relaunch the app (so that the site settings are refreshed), go to Menu --> the POS row should appear shortly
- In wp-admin WC settings, set the currency to non-USD while the country is in the US
- Relaunch the app, go to Menu --> the POS row should **not** appear
- In wp-admin WC settings, set the currency back to USD
- Relaunch the app, go to Menu --> the POS row should appear shortly

## Testing information

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.6

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/POS/POSEligibilityChecker.swift [hunk 1]
```diff
@@ -30,6 +30,9 @@ final class POSEligibilityChecker: POSEligibilityCheckerProtocol {
                 .eraseToAnyPublisher()
         }
         return Publishers.CombineLatest(isWooCommerceVersionSupported, isPointOfSaleFeatureFlagEnabled)
+            .filter { [weak self] _ in
+                self?.isEligibleFromSiteChecks ?? false
```
Comment (staskus) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/POS/POSEligibilityChecker.swift:34]: ❓ Curious

I wonder if `filter` is needed here or not 🤔 I don't remember now what value it provides in `isOnboardingComplete`, do you remember?

**Review by staskus — APPROVED** [review]
Thanks for the improvement, works well 👍

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/POS/POSEligibilityChecker.swift (hunk 1 above)
Comment (jaclync) [reply WooCommerce/Classes/ViewRelated/Dashboard/Settings/POS/POSEligibilityChecker.swift:34]: When the onboarding feature flag is disabled where we still require onboarding completed state for POS, the site checks are performed as a filter as well but on the onboarding state publisher:

https://github.com/woocommerce/woocommerce-ios/blob/e3755bd1a31b26ad19c2bb557895ee5041b077d5/WooCommerce/Classes/ViewRelated/Dashboard/Settings/POS/POSEligibilityChecker.swift#L60-L70

I didn't want to change the behavior there to keep the scope small.

Lemme know if I might have misunderstood your question!

**Pull request merged**